code/datums/elements/supermatter_crystal.dm

```diff
@@ -0,0 +1,343 @@
+/datum/element/supermatter_crystal
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
+	id_arg_index = 2
+
+	var/datum/callback/tool_act_callback
+	var/datum/callback/consume_callback
+
+/datum/element/supermatter_crystal/Attach(datum/target, datum/callback/tool_act_callback, datum/callback/consume_callback)
+	. = ..()
+
+	RegisterSignal(target, COMSIG_ATOM_BLOB_ACT, .proc/blob_hit)
+	RegisterSignal(target, COMSIG_ATOM_ATTACK_PAW, .proc/paw_hit)
+	RegisterSignal(target, COMSIG_ATOM_ATTACK_ANIMAL, .proc/animal_hit)
+	RegisterSignal(target, COMSIG_ATOM_HULK_ATTACK, .proc/hulk_hit)
+	RegisterSignal(target, COMSIG_LIVING_UNARMED_ATTACK, .proc/unarmed_hit)
+	RegisterSignal(target, COMSIG_ATOM_ATTACK_HAND, .proc/hand_hit)
+	RegisterSignal(target, COMSIG_PARENT_ATTACKBY, .proc/attackby_hit)
+	RegisterSignal(target, COMSIG_ATOM_TOOL_ACT(TOOL_WRENCH), .proc/tool_hit)
+	RegisterSignal(target, COMSIG_ATOM_BUMPED, .proc/bumped_hit)
+	RegisterSignal(target, COMSIG_MOVABLE_BUMP, .proc/bump_hit)
+	RegisterSignal(target, COMSIG_ATOM_INTERCEPT_Z_FALL, .proc/intercept_z_fall)
+
+	src.tool_act_callback = tool_act_callback
+	src.consume_callback = consume_callback
+
+/datum/element/supermatter_crystal/Detach(datum/target)
+	var/list/signals_to_remove = list(
```

Comment:
whats the point in making a separate list when you could just do 
`UnregisterSignal(target, list(...))`

Reply:
Readability, I liked it more like this since there are a lot of signals and didn't want to split the proc args